src/bilstm_train_classification.py

```diff
@@ -0,0 +1,255 @@
+
+    def on_epoch_end(self, epoch, logs={}):
+        if epoch % self.interval == 0:
+            y_pred = self.model.predict(self.X_val, verbose=0)
+            score = roc_auc_score(self.y_val, y_pred)
+            print("\n ROC-AUC - epoch: {:d} - score: {:.6f}".format(epoch+1, score))
+
+
+if __name__ == '__main__':
+
+    args = get_args()
+    logger.info(f"Reading test file")
+    test = process_data_classification(args.input_path, args.test_filename)
+    X_test = test["Text-EN"].str.lower()
+    y_test = test['AlexLabel'].values
+    logger.info(f" Using test dataset with {X_test.shape[0]} instances")
+
+    logger.info(f"Reading train file")
+    train = process_data_classification(args.input_path, args.train_filename)
+    X_train = train["Text-EN"].str.lower()
+    y_train = train['AlexLabel'].values
+    logger.info(f" Using train dataset with {X_train.shape[0]} instances")
+
+    tok = get_tokenizer(args)
+    tok.fit_on_texts(list(X_train)+list(X_test))
+    X_train = tok.texts_to_sequences(X_train)
+    X_test = tok.texts_to_sequences(X_test)
+    x_train = sequence.pad_sequences(X_train, maxlen=args.max_length)
+    x_test = sequence.pad_sequences(X_test, maxlen=args.max_length)
+
+    embeddings_index = {}
+    with open(os.path.join(args.embedding_path, args.embedding_file), encoding='utf8') as f:
+        for line in f:
+            values = line.rstrip().rsplit(' ')
+            word = values[0]
+            coefficients = np.asarray(values[1:], dtype='float32')
+            embeddings_index[word] = coefficients
+
+    word_index = tok.word_index
+    # prepare embedding matrix
+    num_words = min(args.max_features, len(word_index) + 1)
+    embedding_matrix = np.zeros((num_words, args.embed_size))
+    for word, i in word_index.items():
+        if i >= args.max_features:
+            continue
+        embedding_vector = embeddings_index.get(word)
+        if embedding_vector is not None:
+            # words not found in embedding index will be all-zeros.
+            embedding_matrix[i] = embedding_vector
+
+    sequence_input = Input(shape=(args.max_length, ))
+    x = Embedding(args.max_features, args.embed_size, weights=[embedding_matrix], trainable=False)(sequence_input)
+    x = SpatialDropout1D(0.2)(x)
+    x = Bidirectional(GRU(128, return_sequences=True, dropout=0.1, recurrent_dropout=0.1))(x)
+    x = Conv1D(64, kernel_size=3, padding="valid", kernel_initializer="glorot_uniform")(x)
+    avg_pool = GlobalAveragePooling1D()(x)
+    max_pool = GlobalMaxPooling1D()(x)
+    x = concatenate([avg_pool, max_pool])
+    preds = Dense(1, activation="sigmoid")(x)
+    model = Model(sequence_input, preds)
+    model.compile(loss='binary_crossentropy', optimizer=Adam(lr=args.learning_rate), metrics=['accuracy'])
+    #model.compile(loss='mse', optimizer=Adam(lr=1e-3), metrics=['accuracy'])
+
+
+
+    X_tra, X_val, y_tra, y_val = train_test_split(x_train, y_train, train_size=0.9, random_state=233)
```

Comment:
I think Javi uses this because he has lots of data so he can see when the network overfits and then validate with the . test set. However for this project is not useful so I will erase it.